### PR TITLE
Feature main activity

### PR DESCRIPTION
**Pull request by tungnt0315**

### CODE REVIEW

File: app/src/main/java/com/example/tung/moviedb_07/data/model/Genre.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import android.os.Parcel;
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class Genre implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/Genre.java]: why we need ``implements Parcelable`` in here?
If you not pass this model to ``Bundle`` or ``Argurment`` Please remove!

File: app/src/main/java/com/example/tung/moviedb_07/data/model/GenreList.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class GenreList implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/GenreList.java]: like above

File: app/src/main/java/com/example/tung/moviedb_07/data/model/Movie.java [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class Movie implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/Movie.java]: like above

File: app/src/main/java/com/example/tung/moviedb_07/data/model/MovieList.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class MovieList implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/MovieList.java]: like above

File: app/src/main/java/com/example/tung/moviedb_07/data/model/ProductionCompany.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import android.os.Parcel;
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class ProductionCompany implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/ProductionCompany.java]: like above

File: app/src/main/java/com/example/tung/moviedb_07/data/model/ProductionCountry.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import android.os.Parcel;
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class ProductionCountry implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/ProductionCountry.java]: like above

File: app/src/main/java/com/example/tung/moviedb_07/data/model/SpokenLanguage.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import android.os.Parcel;
+import android.os.Parcelable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class SpokenLanguage implements Parcelable {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/SpokenLanguage.java]: like above

File: app/src/main/java/com/example/tung/moviedb_07/data/model/VideoList.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Created by tung on 6/8/17.
+ */
+
+public class VideoList implements Parcelable {
+
+    @SerializedName("results")
+    @Expose
+    private List<Video> videos = null;
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/VideoList.java]: remove init data for this variable

File: app/src/main/java/com/example/tung/moviedb_07/screen/list_item/ListItemFragment.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        BaseRecyclerViewAdapter adapter;
+        int tab = getArguments().getInt(Constant.BUNDLE_TAB);
+        if (tab == 4) {
+            adapter = new GenreListAdapter(this.getContext(), null);
+        } else {
+            adapter = new MovieListAdapter(getContext(), null);
+        }
+        Navigator navigator = new Navigator(this);
+        mViewModel = new ListItemViewModel(adapter, navigator);
+        View v = binding.getRoot();
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/screen/list_item/ListItemFragment.java]: ``v`` -> ``view``

File: app/src/main/java/com/example/tung/moviedb_07/screen/list_item/ListItemFragment.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        binding.setViewModel(mViewModel);
+        return v;
+    }
+
+    @Override
+    public void onDetach() {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/screen/list_item/ListItemFragment.java]: Please remove this method and overide ``onStart``, ``onStop`` method
u should refer my sample to develop [link](https://github.com/daolq3012/Structure_Android/blob/mvvm-architecture/app/src/main/java/com/fstyle/structure_android/screen/searchresult/SearchResultActivity.java)

File: app/src/main/java/com/example/tung/moviedb_07/screen/list_item/ListItemViewModel.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    public ListItemViewModel(BaseRecyclerViewAdapter adapter, Navigator navigator) {
+        mAdapter = adapter;
+        mAdapter.setClickListener(this);
+        mNavigator = navigator;
+    }
+
+    @Override
+    public void onItemRecyclerViewClick(Object item) {
+        if (mAdapter instanceof GenreListAdapter) {
+            // code event click genre here.
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/screen/list_item/ListItemViewModel.java]: change to ``// TODO code .....``
add prefix ``TODO`` we can track this on AndroidStudio IDE later

File: app/src/main/java/com/example/tung/moviedb_07/utils/Constant.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+    public static final String API_URL = "https://api.themoviedb.org/3/";
+
+    public static final String API_KEY = "08a61380bb9b09c429ae72d1ae550d5c";
+
+    public static final String IMAGE_URL_ROOT = "http://image.tmdb.org/t/p/w185/";
+
+    public static final String YOUTUBE_URL_ROOT = "https://www.youtube.com/watch?v=";
+
+    public static final String BUNDLE_TAB = "tab";
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/utils/Constant.java]: change values of this constant from ``tab`` -> ``BUNDLE_TAB``

File: app/src/main/res/layout/activity_main.xml [hunk 13]
```diff
@@ -1,21 +1,41 @@
...
+            android:id="@+id/viewPager"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            app:onPageChangeListener="@{(tabLayout)}"
+            app:viewPagerAdapter="@{viewModel.adapter}"
+            />
+
+    </LinearLayout>
+
+</layout>
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_main.xml]: enter when end of file

File: app/src/main/res/layout/item_movie.xml [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+                    android:text="@{viewModel.voteAverage}"
+                    />
+
+            </LinearLayout>
+
+        </LinearLayout>
+
+    </android.support.v7.widget.CardView>
+
+</layout>
```
Comment (daolq3012) [inline app/src/main/res/layout/item_movie.xml]: enter when end of file

File: app/src/main/res/menu/menu_search.xml [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/search_view"
+        android:icon="@android:drawable/ic_menu_search"
+        android:title="@string/search_tittle"
+        app:actionViewClass="android.support.v7.widget.SearchView"
+        app:showAsAction="ifRoom|collapseActionView"/>
+</menu>
```
Comment (daolq3012) [inline app/src/main/res/menu/menu_search.xml]: add empty line when end of file

File: app/src/main/java/com/example/tung/moviedb_07/data/model/Genre.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+
+    @SerializedName("id")
+    @Expose
+    private Integer id;
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/Genre.java]: Pls check name convention of framgia.
variable non-public must have a prefix ``m``

File: app/src/main/java/com/example/tung/moviedb_07/data/model/Genre.java (hunk 1 above)
Comment (tungnt0315) [inline app/src/main/java/com/example/tung/moviedb_07/data/model/Genre.java]: Em định truyền qua Bundle

**Pull request merged**